neuraxle/pipeline.py

```diff
@@ -486,3 +490,222 @@ def should_resume(self, data_container: DataContainer) -> bool:
+        Model()
+    ])
+"""
+
+
+class Barrier(NonFittableMixin, NonTransformableMixin, BaseStep):
+    pass
+
+
+class MiniBatchSequentialPipeline(NonFittableMixin, Pipeline):
+    """
+    Mini Batch Sequential Pipeline class to create a pipeline processing data inputs in batch.
+    """
+
+    def __init__(self, steps: NamedTupleList, batch_size):
+        Pipeline.__init__(self, steps)
+        self.batch_size = batch_size
+
+    def transform(self, data_inputs: Any):
+        """
+        :param data_inputs: the data input to transform
+        :return: transformed data inputs
+        """
+        self.setup()
+
+        current_ids = self.hash(
+            current_ids=None,
+            hyperparameters=self.hyperparams,
+            data_inputs=data_inputs
+        )
+        data_container = DataContainer(current_ids=current_ids, data_inputs=data_inputs)
+        data_container = self.handle_transform(data_container)
+
+        return data_container.data_inputs
+
+    def fit_transform(self, data_inputs, expected_outputs=None) -> ('Pipeline', Any):
+        """
+        :param data_inputs: the data input to fit on
+        :param expected_outputs: the expected data output to fit on
+        :return: the pipeline itself
+        """
+        self.setup()
+
+        current_ids = self.hash(
+            current_ids=None,
+            hyperparameters=self.hyperparams,
+            data_inputs=data_inputs
+        )
+        data_container = DataContainer(
+            current_ids=current_ids,
+            data_inputs=data_inputs,
+            expected_outputs=expected_outputs
+        )
+        new_self, data_container = self.handle_fit_transform(data_container)
+
+        return new_self, data_container.data_inputs
+
+    def handle_transform(self, data_container: DataContainer) -> DataContainer:
+        """
+        Transform all sub pipelines splitted by the Barrier steps.
+
+        :param data_container: data container to transform.
+        :return: data container
+        """
+        sub_pipelines = self._create_sub_pipelines()
+
+        for sub_pipeline in sub_pipelines:
+            data_container = self._handle_transform_sub_pipeline(sub_pipeline, data_container)
+
+        return data_container
+
+    def handle_fit_transform(self, data_container: DataContainer) -> \
+            Tuple['MiniBatchSequentialPipeline', DataContainer]:
+        """
+        Transform all sub pipelines splitted by the Barrier steps.
+
+        :param data_container: data container to transform.
+        :return: data container
+        """
+        sub_pipelines = self._create_sub_pipelines()
+
+        for sub_pipeline in sub_pipelines:
+            new_self, data_container = self._handle_fit_transform_sub_pipeline(sub_pipeline, data_container)
+
+        return self, data_container
+
+    def _handle_transform_sub_pipeline(self, sub_pipeline, data_container) -> DataContainer:
+        """
+        Transform sub pipeline using join transform.
```

Comment:
This method might move into the joiner to allow for different joiners to handle as they wish to handle.